src/test/java/BaseTest.java

```diff
@@ -63,6 +69,36 @@ public void launchBrowser(@Optional String baseURL) {
 
     }
 
+    private WebDriver pickBrowser() throws MalformedURLException {
+        DesiredCapabilities capabilities = new DesiredCapabilities();
+        String gridURL = "http://192.168.0.20:4444";
+
+        String browser = System.getProperty("browser");
+        WebDriver webDriver = null;
+
+        switch (browser) {
+            case "firefox":
+                System.setProperty("webdriver.gecko.driver", "geckodriver.exe");
+                webDriver = new FirefoxDriver();
+                break;
+            case "grid-firefox":
+                capabilities.setCapability("browserName", "firefox");
+                webDriver = new RemoteWebDriver(URI.create(gridURL).toURL(), capabilities);
+                break;
+            case "grid-chrome":
+                capabilities.setCapability("browserName", "chrome");
+                webDriver = new RemoteWebDriver(URI.create(gridURL).toURL(), capabilities);
+                break;
+            case "edge":
+                webDriver = new EdgeDriver();
+                break;
+            default:
+                webDriver = new ChromeDriver();
+        }
+
+        return webDriver;
+    }
+
```

Comment:
**Step 1**  Add "geckodriver.exe"
For example
`System.setProperty("webdriver.gecko.driver","geckodriver.exe");`
![image](https://user-images.githubusercontent.com/104761482/201294239-cb6ccd25-6531-405c-ac9b-bb2fc09cbdda.png)


**Step 2**  Write a method to initialize the driver, for example
`driver = pickBrowser("firefox");`
![image](https://user-images.githubusercontent.com/104761482/201294328-fab7a7af-3f50-49ea-88fb-e990d206cce6.png)

**1. To start the selenium server:**

For example java `-jar selenium-server-4.6.0.jar standalone`
![image](https://user-images.githubusercontent.com/104761482/201294494-6de41193-36c9-4df5-8dd1-240fe3138989.png)

2. Add an URL to your project, for example

![image](https://user-images.githubusercontent.com/104761482/201294734-4c663d2b-0dbc-42d7-98fe-2b8975d3b66a.png)

`String gridURL = "http://172.21.160.1:4444/";`

![image](https://user-images.githubusercontent.com/104761482/201294845-e5f2b224-e97b-4dae-b98c-fd7928cd4e2e.png)

![image](https://user-images.githubusercontent.com/104761482/201294920-0f93d1b0-7c18-4c04-bd5d-b8193cb8ebbb.png)

___________________________________________________________________________________________________________________________

```
public class BaseTest {

  protected WebDriver driver;
  protected String url;
  protected WebDriverWait wait;
  protected Actions actions;


  @BeforeSuite
  public static void chromeConfigs() {
    if (System.getProperty("os.name").toLowerCase().contains("win")) {
      System.setProperty("webdriver.chrome.driver", "chromedriver.exe");
    }
  }

  @DataProvider(name = "invalidCredentials")
  public static Object[][] getCredentials() {

    return new Object[][]{
        {"invalid@class.com", "invalidPass"},
        {"d@class.com", ""},
        {"", ""}
    };
  }

  @BeforeMethod
  @Parameters({"baseURL"})
  public void launchBrowser(@Optional String baseURL) throws MalformedURLException {
    if (baseURL == null) {
      baseURL = "https://bbb.testpro.io";
    }
    driver = pickBrowser("firefox");
    actions = new Actions(driver);
    driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(5));
    wait = new WebDriverWait(driver, Duration.ofSeconds(10));
    url = baseURL;
    driver.get(url);

  }

  public WebDriver pickBrowser(String browser) throws MalformedURLException {
    DesiredCapabilities capabilities = new DesiredCapabilities();
    String gridURL = "http://172.21.160.1:4444/";
    switch (browser) {
      case "firefox":
        System.setProperty("webdriver.gecko.driver", "geckodriver.exe");
        return driver = new FirefoxDriver();
      case "safari":
        return driver = new SafariDriver();
      case "grid-safari":
        capabilities.setCapability("browserName", "safari");
        return driver = new RemoteWebDriver(URI.create(gridURL).toURL(), capabilities);
      case "grid-firefox":
        capabilities.setCapability("browserName", "firefox");
        return driver = new RemoteWebDriver(URI.create(gridURL).toURL(), capabilities);
      case "grid-chrome":
        capabilities.setCapability("browserName", "chrome");
        return driver = new RemoteWebDriver(URI.create(gridURL).toURL(), capabilities);
      default:
        return driver = new ChromeDriver();

    }
  }

  @AfterMethod
  public void tearDownBrowser() {
    driver.quit();
  }

  public void clickSubmitBtn() {
    WebElement submitButton = driver.findElement(By.cssSelector("[type='submit']"));
    submitButton.click();
  }

  public void provideEmail(String email) {
    WebElement emailField = driver.findElement(By.cssSelector("[type='email']"));
    emailField.click();
    emailField.sendKeys(email);
  }

  public void providePassword(String password) {
    WebElement passwordField = driver.findElement(By.cssSelector("[type='password']"));
    passwordField.click();
    passwordField.sendKeys(password);

  }

  public void login() {
    provideEmail("demo@class.com");
    providePassword("te$t$tudent");
    clickSubmitBtn();
  }
}

```

**Please, run tests and check**